torch/_functorch/_aot_autograd/subclass_utils.py

```diff
@@ -79,12 +79,27 @@ def create_subclass_meta(
 def unwrap_tensor_subclasses(wrapped_args, *, is_joint_structure: bool):
```

Comment:
@guilhermeleobas let me know if you want to chat more - I think I would expect:

(1) the symints of any subclass tensor inputs become extra inputs to the graph, allowing `make_fx()` to "see" them during tracing - what this PR currently does

(2) the symints of any subclass output tensors becomes extra outputs to the graph (this is what will allow any compute on the subclass symints to show up inside of the graph so we can compile it - I don't currently see this)

(3) when we wrap up outputs at runtime back into tensor subclasses, we use the symint outputs that were returned to do so (I don't currently see this)

Reply:
Hi Brian,

I tried a few approaches before finalizing the code I committed.

My strategy involved replacing the SymInts in the wrapped_args list with those from the tensor subclass. Since the original SymInts are unusable and any attempt to use them triggers a `RuntimeError: s0 is not tracked with proxy for PythonKeyTracer object`, replacing them with the newer SymInts resolves the proxy issue. This way, we don't have to change the function since the number of arguments remains the same.

> the symints of any subclass output tensors becomes extra outputs to the graph (this is what will allow any compute on the subclass symints to show up inside of the graph so we can compile it - I don't currently see this)

Do you have an example of a function that the symints of the subclass output would show up in the graph?

> (3) when we wrap up outputs at runtime back into tensor subclasses, we use the symint outputs that were returned to do so (I don't currently see this)

Yeah, I forgot about this one.

Reply:
For (2), the following should be an example

```python
@torch.compile(backend='aot_eager', dynamic=True)
def fn(x):
    y = x.clone()
    return y.view(y.shape[0] * y.shape[1])
```

Reply:
I think I fixed all the issues you listed, @bdhirsh. Just one small assertion that I've commented before that I need to re-enable.